### ISSUE TITLE
re2s needs reduced code size

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#1573]
Per discussions with @densh:

The recent re2s regex implementation used two dimensional arrays for Unicode tables
because that is what was in the scala code being ported.  This allowed the
substitution of re2s for cre2 but with known IOUs.

The mid range plan is to unify the Unicode tables in Character & re2s.

There is immediate benefit in both reduced code size reduction and increased access speed
to use the one dimensional tables that were in the original original C code.
Time has come to pay one IOU.

**Comment by densh** [comment]
@LeeTibbert Any luck with this one? I am hoping to cut 0.4.0-M1 early next week and we would be nice to have this fixed as well but it's not a blocker (it can wait until M2).

**Comment by LeeTibbert** [comment]
I am actively working on this one. Like most things, it is not as simple as
it looks/sounds.  What can be so hard about converting 2-D to 1-D and
creating a few access methods in anticipation of Unicode unification?
A day's work, including the writeup.

tl;dr  -  Probably should schedule for M2.
At the start of the day yesterday, 3 Suites failed, the important one.
I made fixes which got one working, so I know that at least some of
my tables and accessor methods work.  

Two Suites remain. One of those is the problem I was chasing 10 days
or so where the remote test runner hangs before it can connect back
to the local master.   I am hoping that fixing the simpler of the two
fixes that.

If things fall into place before M1 I will let you know.  I want to 
exercise this one quite well before sending it into the wild.

Sound like a plan?

**Comment by densh** [comment]
Sure, we can do it in M2.
